src/packaging/kpatch/commit_to_overlay.sh

```diff
@@ -445,9 +448,17 @@ for change in ${COMMIT_SHAS[@]}; do
   echo "Generating the kernel overlay"
   cd src/packaging/kpatch
 
-  export GITDATE="${gitdate}" GITAUTHOR="${gitauthor}" COMMIT_FILE="${commit_file}" \
-         LINUX_SRC=${LINUX_SRC}
-  make prepare kernel-overlay >> ${build_log_file}
+  export LINUX_SRC=${LINUX_SRC}
+  make prepare >> ${build_log_file}
+
+  WORK_DIR=work/kvdo-* # Location of the prepared kernel products
+  rm -rf ${LINUX_VDO_SRC}
+  mkdir -p ${LINUX_VDO_SRC}
+  cp -r ${WORK_DIR}/dm-vdo/* ${LINUX_VDO_SRC}
+  sed -i -E -e 's/(#define	CURRENT_VERSION).*/\1 "8.3.0.65"/' \
+    ${LINUX_VDO_SRC}/dm-vdo-target.c
+  mv ${LINUX_VDO_SRC}/dm-vdo-target.c ${LINUX_MD_SRC}
+  cp -f ${WORK_DIR}/vdo.rst ${WORK_DIR}/vdo-design.rst ${LINUX_DOC_SRC}/
 
   if [[ $? != 0 ]]; then
```

Comment:
Point taken. And I agree with the principle. How should I spell this? I could divert the output of each command to the build log. If I create a function to encapsulate the overlay steps, will it automatically return the command output, or would I have to do something else to make that happen?

On another hand, these steps are all fairly simple commands; is it enough just to have to script report any errors directly as script output? I'm not sure there is enough content to even need to divert it to another log file.

Anyway. I'm happy to change this, but I could use some more specific guidance about what better looks like.

Reply:
Redirecting stderr and stdout for each command to the build_log_file would be sufficient. 

Along with that, maybe the best thing is to create a trap function to trap errors. For instance, at the start of overlay generation, have `trap generate_overlay_failed ERR`. This would trap any error in the generate steps until the end where a `trap - ERR` would be needed to unset the trap. Then you would only need to create create the generate_overlay_failed function and put the error message from line 464 in there.